src/components/Identity/createComponent.js

```diff
@@ -74,61 +56,54 @@ export default (idSyncs, config, logger, cookieJar, optIn, eventManager) => {
-            logger.log("Delaying request while retrieving ECID from server.");
-            promise = deferredForEcid.promise.then(() => {
-              logger.log("Resuming previously delayed request.");
-              addIdsContext(payload, getEcid());
-            });
-          } else {
-            // We don't have an ECID and no request has gone out to fetch it.
-            // We won't apply the ECID to this request, but we'll set up a
-            // promise so that future requests can know when the ECID has returned.
-            deferredForEcid = defer();
-            payload.expectResponse();
-            if (
-              config.thirdPartyCookiesEnabled &&
-              areThirdPartyCookiesSupported(getBrowser(window))
-            ) {
-              payload.useIdThirdPartyDomain();
+          if (!hasEcid()) {
+            if (deferredForEcid) {
+              // We don't have an ECID, but the first request has gone out to
+              // fetch it. We must wait for the response to come back with the
+              // ECID before we can apply it to this payload.
+              logger.log("Delaying request while retrieving ECID from server.");
+              promise = deferredForEcid.promise.then(() => {
+                logger.log("Resuming previously delayed request.");
+              });
+            } else {
+              // We don't have an ECID and no request has gone out to fetch it.
+              // We won't apply the ECID to this request, but we'll set up a
+              // promise so that future requests can know when the ECID has returned.
+              deferredForEcid = defer();
+              payload.expectResponse();
+              // If third-party cookies are enabled by the customer and
+              // supported by the browser, we will send the request to a
+              // a third-party domain that allows for more accurate
+              // identification of the user through use of a third-party cookie.
+              if (
+                config.thirdPartyCookiesEnabled &&
+                areThirdPartyCookiesSupported(getBrowser(window))
+              ) {
+                payload.useIdThirdPartyDomain();
+              }
             }
           }
+
           customerIds.addToPayload(payload);
           return promise;
         });
       },
 
-      // Waiting for opt-in because we'll be writing the ECID to a cookie
-      onResponse({ response }) {
+      // Waiting for opt-in because we'll be reading the ECID from a cookie
+      onResponse() {
         return optIn.whenOptedIn().then(() => {
-          const ecidPayloads = response.getPayloadsByType("identity:persist");
-
-          if (ecidPayloads.length > 0) {
-            const ecid = ecidPayloads[0].id;
-            cookieJar.set(EXPERIENCE_CLOUD_ID, ecid);
-            migration.createAmcvCookie(ecid);
-            if (deferredForEcid) {
-              deferredForEcid.resolve();
-            }
+          if (deferredForEcid && hasEcid()) {
```

Comment:
Relying on the set cookies here seems risky.  

Scenario 1. Konductor stops sending the expected cookie name.
Scenario 2. Safari implements a new ITP and blocks first party cookies set from the browser.  

In these scenarios, we would be dropping the additional requests from the page.  I think most customers would rather create additional visits than drop events.

Reply:
I think you're probably right. I believe you're suggesting to remove the `hasEcid()` check here. @jfkhoury maybe you know more about what the customer would want? I can bring it up in our meeting with PM tomorrow.

Reply:
We discussed these scenarios with Erin yesterday and she indicated that she would rather not have hits being sent without an ECID.